README-METRICS.md

```diff
@@ -1 +1,94 @@
+
+```bash
+make remote-vanilla
+```
+
+### What's Not Great About the VanillaFunction Metrics?
+
+There is nothing inherently wrong about the metrics that are collected by CloudWatch Metrics. They allow us to view invocations, errors, and performance data. For application operators needing to understand Lambda function behavior, this is likely sufficient.
+
+![Standard Lambda Metrics](./img/metrics/vanilla-function-metrics-lambda-console.png)
+
+However, our function is not emitting business-specific metrics. It is likely our application has business stakeholders who want to understand the business value being driven by our function. Fortunately, we can add this capability easily.
+
+## Metrics - Instrumented Edition
+
+The `InstrumentedFunction` Lambda function emits a custom metric, relevant to our fictional user registration system. Business users have a need to understand when user registrations are successful and when they fail.
+
+Let's invoke `InstrumentedFunction` in the AWS cloud with the following command:
+
+```bash
+make remote-instrumented
+```
+
+We can then view the output generated in the CloudWatch Metrics console:
+
+![Custom Lambda Metrics - User Registrations](./img/metrics/instrumented-function-metrics-console-1.png)
+
+### How Did We Do That?
+
+Developers can publish custom metrics asynchronously to CloudWatch Logs using a defined schema - [Embedded Metrics Format (EMF)](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html). Logs that conform to the EMF schema are published automatically to CloudWatch Metrics. While metrics can be published to CloudWatch Metrics using the [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudwatch/client/put_metric_data.html) module, fight the urge. The `put_metric_data` method is synchronous in nature. Any latency or errors publishing to CloudWatch Metrics will result in latency for your users and higher costs in both function execution and CloudWatch API calls. There is no extra cost to use EMF beyond log storage.
+
+How did we capture this information?
+
+In our example, we again use the [Powertools for AWS Lambda (Python)](https://docs.powertools.aws.dev/lambda/python/latest/) module. If the HTTP response is a `200` in our `submit_user` function then [instrumented.py](./function/instrumented/instrumented.py#L43) will add 1 to `SuccessfulSubmission`. If the HTTP response is not a `200` then [instrumented.py](./function/instrumented/instrumented.py#L45) will add 1 to `UnsuccessfulSubmission`.
+
+Could We Use AWS' EMF Python Module?
+
+Absolutely! The [aws-embedded-metrics module](https://github.com/awslabs/aws-embedded-metrics-python) is available for use and allows developers to publish metrics using EMF. However, Powertools for AWS Lambda (Python) makes this really easy. In general, you should use Powertools for AWS Lambda (Python) for your custom metrics needs.
+
```

Comment:
I particularly like this from a leveling up perspective.

Reply:
Thanks much, @djfurman!